File: golang_app/presenter/sportstatlabels.go
package presenter

import (
	"github.com/asma12a/challenge-s6/ent/schema/ulid"
)


type SportStatLabels struct {
	ID         ulid.ID          `json:"id,omitempty"`
	Label      string           `json:"label,omitempty"`
	Unit       string           `json:"unit,omitempty"`
	IsMain     bool             `json:"is_main,omitempty"`
	Sport      *Sport            `json:"sport,omitempty"` 
}

File: flutter_app/lib/core/models/sport_stat_labels.dart
import 'dart:ffi';

import 'package:squad_go/core/models/sport.dart';

class SportStatLabels {

  final String? id;
  final String label;
  final String? unit;
  final bool? isMain;

  SportStatLabels({
    this.id,
    required this.label,
    this.unit,
    this.isMain
  });



  factory SportStatLabels.fromJson(Map<String, dynamic> json) {
    return SportStatLabels(
      id: json['id'],
      label: json['label'],
      unit: json['unit'],
      isMain: json['is_main'] ?? false,
    );
  }

}

